client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,634 @@
+            // Stay in Connecting, could retry with different leader
+          } yield ()
+          
+        case SessionNotFound =>
+          // Session doesn't exist, stay in Connecting and retry
+          ZIO.unit
+          
+        case _ =>
+          // Other errors - transition to Disconnected
+          client.transitionTo(Disconnected)
+      }
+  }
+  
+  /**
+   * Client has an active session and can send/receive requests.
+   * Contains all session-related data.
+   */
+  case class Connected(
+    sessionId: SessionId,
+    capabilities: Map[String, String],
+    createdAt: Instant,
+    serverRequestTracker: ServerRequestTracker
+  ) extends ClientState {
+    override def stateName: String = s"Connected($sessionId)"
+    
+    override def handleMessage(message: ServerMessage, client: RaftClient): UIO[Unit] = 
+      message match {
+        case response: ClientResponse =>
+          handleClientResponse(response, client)
+        
+        case KeepAliveResponse(timestamp) =>
+          ZIO.logDebug(s"Keep-alive acknowledged: $timestamp")
+        
+        case serverRequest: ServerRequest =>
+          handleServerRequest(serverRequest, client)
+        
+        case error: RequestError =>
+          handleRequestError(error, client)
+        
+        case SessionClosed(reason, leaderId) =>
+          for {
+            _ <- ZIO.logInfo(s"Session closed by server: $reason")
+            _ <- client.transitionTo(Disconnected)
+          } yield ()
+        
+        case other =>
+          ZIO.logWarning(s"Unexpected message in Connected state: ${other.getClass.getSimpleName}")
+      }
+    
+    private def handleClientResponse(response: ClientResponse, client: RaftClient): UIO[Unit] = 
+      for {
+        // Complete pending request via retry manager
+        _ <- client.retryManager.completePendingRequest(response.requestId, response.result)
+      } yield ()
+    
+    private def handleServerRequest(request: ServerRequest, client: RaftClient): UIO[Unit] = 
+      for {
+        _ <- ZIO.logInfo(s"Received server request: ${request.requestId}")
+        // User should call acknowledgeServerRequest after processing
+      } yield ()
+    
+    private def handleRequestError(error: RequestError, client: RaftClient): UIO[Unit] = 
+      error.reason match {
+        case NotLeaderRequest =>
+          for {
+            _ <- ZIO.logInfo(s"Not leader, redirect to: ${error.leaderId}")
+            // Could update leader information and retry
+          } yield ()
+        
+        case SessionTerminated =>
+          for {
+            _ <- ZIO.logWarning("Session terminated by server")
+            _ <- client.transitionTo(Disconnected)
+          } yield ()
+        
+        case _ =>
+          ZIO.logWarning(s"Request error: ${error.reason}")
+      }
+  }
+}
+
+/**
+ * Pending session operations tracked by nonce.
+ */
+private sealed trait SessionOperation
```

Comment:
we don't need this. The nonce should just be a parameter to the Connecting state, if it receive a reply with different nonce it would just ignore it.